modules/apps/headless/headless-object/headless-object-impl/src/main/java/com/liferay/headless/object/internal/resource/v1_0/CollaboratorResourceImpl.java

```diff
@@ -18,4 +53,350 @@
+			_objectEntryLocalService.getObjectEntry(
+				externalReferenceCode, contextCompany.getCompanyId(),
+				_getGroupId(scopeKey)),
+			pagination);
+	}
+
+	@Override
+	public Page<Collaborator>
+			getScopeScopeKeyObjectEntryFolderByExternalReferenceCodeCollaboratorsPage(
+				String scopeKey, String externalReferenceCode,
+				Pagination pagination)
+		throws Exception {
+
+		if (!FeatureFlagManagerUtil.isEnabled("LPD-17564")) {
+			throw new UnsupportedOperationException();
+		}
+
+		return _getObjectEntryFolderCollaboratorsPage(
+			_objectEntryFolderLocalService.
+				getObjectEntryFolderByExternalReferenceCode(
+					externalReferenceCode, _getGroupId(scopeKey),
+					contextCompany.getCompanyId()),
+			pagination);
+	}
+
+	@Override
+	public Page<Collaborator> postObjectEntryCollaboratorsPage(
+			Long objectEntryId, Collaborator[] collaborators)
+		throws Exception {
+
+		if (!FeatureFlagManagerUtil.isEnabled("LPD-17564")) {
+			throw new UnsupportedOperationException();
+		}
+
+		return _postObjectEntryCollaboratorsPage(
+			collaborators,
+			_objectEntryLocalService.getObjectEntry(objectEntryId));
+	}
+
+	@Override
+	public Page<Collaborator> postObjectEntryFolderCollaboratorsPage(
+			Long objectEntryFolderId, Collaborator[] collaborators)
+		throws Exception {
+
+		if (!FeatureFlagManagerUtil.isEnabled("LPD-17564")) {
+			throw new UnsupportedOperationException();
+		}
+
+		return _postObjectEntryFolderCollaboratorsPage(
+			collaborators,
+			_objectEntryFolderLocalService.getObjectEntryFolder(
+				objectEntryFolderId));
+	}
+
+	@Override
+	public Page<Collaborator>
+			postScopeScopeKeyObjectEntryByExternalReferenceCodeCollaboratorsPage(
+				String scopeKey, String externalReferenceCode,
+				Collaborator[] collaborators)
+		throws Exception {
+
+		if (!FeatureFlagManagerUtil.isEnabled("LPD-17564")) {
+			throw new UnsupportedOperationException();
+		}
+
+		return _postObjectEntryCollaboratorsPage(
+			collaborators,
+			_objectEntryLocalService.getObjectEntry(
+				externalReferenceCode, contextCompany.getCompanyId(),
+				_getGroupId(scopeKey)));
+	}
+
+	@Override
+	public Page<Collaborator>
+			postScopeScopeKeyObjectEntryFolderByExternalReferenceCodeCollaboratorsPage(
+				String scopeKey, String externalReferenceCode,
+				Collaborator[] collaborators)
+		throws Exception {
+
+		if (!FeatureFlagManagerUtil.isEnabled("LPD-17564")) {
+			throw new UnsupportedOperationException();
+		}
+
+		return _postObjectEntryFolderCollaboratorsPage(
+			collaborators,
+			_objectEntryFolderLocalService.
+				getObjectEntryFolderByExternalReferenceCode(
+					externalReferenceCode, _getGroupId(scopeKey),
+					contextCompany.getCompanyId()));
+	}
+
+	private SharingEntry _addOrUpdateSharingEntry(
+			long classNameId, long classPK, long groupId,
+			Collaborator collaborator)
+		throws Exception {
+
+		long toUserId = 0;
+		long toUserGroupId = 0;
+
+		if (StringUtil.equals(collaborator.getType(), "user")) {
```

Comment:
@Mikellorza, here you already have the type, cannot we use this better in the DTO as mentioned before?

Reply:
I have the type fpr the DTO, since we have the externalReferenceCode and we need to distinguish between a User and a UserGroup ,but, I don't have that type in the SharingEntry model. In the SharingEntry model I have two fields, one for toUserId referencing to the userId and that means the entry has been shared to the user directly, and the toUserGroupId that is referencing to the userGroupId  and that means the entry has been shared to the user group.

Reply:
Same as the other comments, we can use the enum that is generated

Reply:
I changed it, thanks mozo